flink-runtime/src/main/java/org/apache/flink/runtime/metrics/util/MetricUtils.java

```diff
@@ -338,6 +339,30 @@ static void instantiateMetaspaceMemoryMetrics(final MetricGroup parentMetricGrou
         }
     }
 
+    static void instantiateFileDescriptorMetrics(MetricGroup metrics) {
+        try {
+            final com.sun.management.OperatingSystemMXBean mxBean =
+                    (com.sun.management.OperatingSystemMXBean)
+                            ManagementFactory.getOperatingSystemMXBean();
+
+            if (mxBean instanceof com.sun.management.UnixOperatingSystemMXBean) {
```

Comment:
If the class is not found then wouldn't `instanceof` fail ?

Reply:
Yes, it will throw exception and caught in line 358.

Reply:
> Yes, it will throw exception and caught in line 358.

Ok thanks. In the case where the instanceof fails it would be good to mention the that JVM does not have this class in the error message, maybe a separate catch to give a more informative message. 

How do we get into the else case? Can a bean of the wrong type get into this code?

Reply:
The target java version of current main branch of Flink is java 17. The class com.sun.management.UnixOperatingSystemMXBean exist since all Java 1.5. 
https://docs.oracle.com/en/java/javase/11/docs/api/jdk.management/com/sun/management/UnixOperatingSystemMXBean.html.

But whether UnixOperatingSystemMXBean can be returned, depends on underneath OS. It user run the flink code in window locally, I guess UnixOperatingSystemMXBean will not be created. Thus, we throw exception in this case. In the end, the error log will print the exception stack.